skbio/metadata/_interval.py

```diff
@@ -0,0 +1,334 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from ._intersection import IntervalTree
+
+
+class Interval():
+    '''Stores the position and metadata of an interval.
+
+    Parameters
+    ----------
+    intervals : list of tuple of ints
+        List of tuples representing start and end coordinates.
+    boundaries : list of tuple of bool
```

Comment:
`iterable` instead of `list` (add unit test to make sure it works with an iterable)